Sources/SentryCrash/Recording/Monitors/SentryCrashMonitor_MachException.c

```diff
@@ -313,7 +313,30 @@ handleExceptions(void *const userData)
         kern_return_t kr = mach_msg(&exceptionMessage.header, MACH_RCV_MSG, 0,
             sizeof(exceptionMessage), g_exceptionPort, MACH_MSG_TIMEOUT_NONE, MACH_PORT_NULL);
         if (kr == KERN_SUCCESS) {
-            break;
+            // If the exception was generated in a different process than the one in which
+            // the handler was installed, ignore it.  Task-level exception handlers are
+            // inherited by subprocesses, and we don't want to handle or report exceptions
+            // in arbitrary subprocesses of the one in which we were installed.
+            if (exceptionMessage.task.name != mach_task_self()) {
+                SentryCrashLOG_DEBUG("Received exception from subprocess. Ignoring message and waiting for another.");
+                // Send a reply saying "I didn't handle this exception".
```

Comment:
There is similar code later in this function which claims to reply and say "I didn't handle this exception".  I initially tried using that code here, but it did not appear to work correctly - subprocesses would not terminate post-exception as one would have expected.

Reply:
Thanks a lot for opening up this draft PR, @vorporeal 🚀. At a first glance, this PR makes sense. @armcknight I think you have more knowledge on mach messages. Could you please have a look?